pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -508,8 +478,8 @@ func (n *NetworkPolicyController) createAppliedToGroup(npNsName string, pSel, nS
 // NetworkPolicyPeer object in NetworkPolicyRule. This function simply
 // creates the object without actually populating the PodAddresses as the
 // affected GroupMembers are calculated during sync process.
-func (n *NetworkPolicyController) createAddressGroup(namespace string, podSelector, nsSelector, eeSelector *metav1.LabelSelector) string {
-	groupSelector := toGroupSelector(namespace, podSelector, nsSelector, eeSelector)
+func (n *NetworkPolicyController) createAddressGroup(namespace string, podSelector, nsSelector, eeSelector *metav1.LabelSelector, nodeSelectors ...*metav1.LabelSelector) string {
+	groupSelector := antreatypes.NewGroupSelector(namespace, podSelector, nsSelector, eeSelector, nodeSelectors...)
```

Comment:
Please note the impact of adding addressGroup with nodeSelector only to `groupingInterface`, I'm not sure how would it behave since it expects there is at least podSelector or eeSelector.

Reply:
I think the test cases have covered the situations that `adding addressGroup with nodeSelector only to groupingInterface`? Maybe is there any other situation that I didn’t consider?

Reply:
I think if the addressGroup has nodeSelector only, we shouldn't add it to groupingInterface to avoid unncessary overhead, as we never consume its result. Otherwise it would still be processes when matching any Pods.